frontend/src/views/admin/matching/index.tsx

```diff
@@ -1,14 +1,153 @@
+    let combinedApplication: Application | null = null;
+    for (const posting of Object.keys(applicationsByPosting)) {
+        applicationsByPosting[+posting].sort((a, b) => {
+            if (a.submission_date === b.submission_date) {
+                return 0;
+            }
+            if (a.submission_date > b.submission_date) {
+                return 1;
+            }
+            return -1;
+        });
+
+        const newestApplication =
+            applicationsByPosting[+posting][
+                applicationsByPosting[+posting].length - 1
+            ];
+        if (!combinedApplication) {
+            combinedApplication = newestApplication;
+        } else {
+            // Update values to include info from this application
+            for (const positionPref of newestApplication.position_preferences) {
+                const matchingPref =
+                    combinedApplication.position_preferences.find(
+                        (pref) =>
+                            pref.position.position_code ===
+                            positionPref.position.position_code
+                    ) || null;
+
+                if (!matchingPref) {
+                    combinedApplication.position_preferences.push(positionPref);
+                } else {
+                    // Update position preference to take the maximum of the two
+                    matchingPref.preference_level = Math.max(
+                        positionPref.preference_level,
+                        matchingPref.preference_level
+                    );
+                }
+            }
+
+            // Combine custom question answers?
+        }
+    }
+
+    return combinedApplication;
+}
 
 export function AdminMatchingView() {
     const activeSession = useSelector(activeSessionSelector);
     const dispatch = useThunkDispatch();
 
+    const [selectedPosition, setSelectedPosition] =
+        React.useState<PositionSummary | null>(null);
+    const [updated, setUpdated] = React.useState(false);
+
+    const positions = useSelector(positionsSelector);
+    const assignments = useSelector(assignmentsSelector);
+    const applications = useSelector(applicationsSelector);
+    const applicants = useSelector(applicantsSelector);
+    const matches = useSelector(matchesSelector);
+    const guarantees = useSelector(guaranteesSelector);
+    const notes = useSelector(notesSelector);
+    const matchingData = useSelector(matchingDataSelector);
+
+    const [localStore, setLocalStore] = React.useState(() => {
+        const saved = localStorage.getItem("matchingData");
+        if (!saved) {
+            return {
+                matches: [],
+                guarantees: [],
+                notes: {},
+            } as MatchingDataState;
+        }
+
+        return JSON.parse(saved) as MatchingDataState;
+    });
+
+    React.useEffect(() => {
+        localStorage.setItem("matchingData", JSON.stringify(localStore));
+    }, [localStore]);
```

Comment:
This is not the right way to do this. This should be handled by a Redux thunk. You should create a high-level function that takes a thunk/action, and returns a thunk that performs the action and then saves at the end. E.g.

```typescript
autoSaveWrapper = async (thunk, { dispatch, getState }) => { await dispatch(thunk); saveToLocalStorage(matchingDataSelector(getState()))}
```

Then, in the other components you can do `dispatch(autoSaveWrapper(updateMatch(...)))`. The saving will happen automagically.

(PS: I don't have the function signatures quite right, so the actual functions will look different)